Data Portal Charter

theathorn opened the pull request:
~~**October 19**: Last call for community reviews~~
~~**October 26**: Last call for oversight reviews~~
**November 15**: Approved during DCP Architecture call

theathorn committed:
Create charter.md

theathorn committed:
Initial draft

theathorn committed:
Incorporate initial feedback from Brian O'Connor

briandoconnor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,82 @@
+
+## In-scope
+### General
+* UX, visual and branding design with input from CZI and the GG
```
maybe you want to say "leadership (GG, scientific steering committee, etc)"

theathorn replied:
done

briandoconnor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,82 @@
+## In-scope
+### General
+* UX, visual and branding design with input from CZI and the GG
+* Production operation of the overall Data Portal web-site (see exclusions in Out-of-scope section)
```
Say anything about participating in integration testing, CI, CD, centralized logging system, etc?

theathorn replied:
Added

briandoconnor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,82 @@
+### Data Portal
+* Select and deploy content management system
+* Integrate Data Browser and Data Portal UIs into a seamless UX and provide link to other groups' applications (e.g. Ingest UI)
+* Frequent updates of summary statistics (cells, organs, donors, projects, labs, etc.) on Home Page
```
Should be live since it's event driven (within 30 seconds or so) but don't worry too much about that level of detail here.

theathorn replied:
OK

briandoconnor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,82 @@
+* Coordinate development of user guide content (e.g. “Learn”) from multiple contributors
+* Coordinate development of developer documentation content (e.g. “Develop”) from multiple contributors. Includes API documentation and development guides.
+* Click through to Data Browser from search box and human body image
+* Registration form for tertiary portals and list of portals available with click through to hosting web site (e.g. Dockstore)
```
And eventually this registration process will include structured "handoff" information, so in the future tertiary analysis portals will be able to receive search results from the Data Browser.

theathorn replied:
See Data Browser section under "Handoff of a Collection..."

briandoconnor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,82 @@
+* Integration with Canny (to capture user feedback)
+### Data Browser
+* Create (or use Data Store provided) indexes as necessary for files in the Data Store, using public DSS APIs to read the metadata
+* Browser support for multiple metadata schemas
```
Might want to say the indexer is using the metadata API developed by Hannes... in which case we support schema versions that that API (maintained by purple) supports

theathorn replied:
Updated

briandoconnor commented:
Hi level comments/questions that I had as I went through this. Feel free to accept or reject... and if Brian R doesn't think they should be addressed in a charter that's fine too.  

* probably want to specifically mention how much time is covered by this charter... does this charter cover the next year?

**[Brian]** The milestones should cover the time. 

* probably want to give an overview of the approximate number of people and their roles, since that helps to give some context on why we're proposing these responsibilities.

**[Brian]** No need to include an orgchart. If there are concerns about staffing, these could be discussed during review by the technical leads.

briandoconnor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,82 @@
+* AuthN for tracking users' collections, saved searches, etc.
+* Full authN/Z if the HCA stores controlled access data in the future. Leverage AuthN/Z from Data Store.
+* Browser interface with DSS Collections service to save/view/list/share/delete the query results from a search operation
+* Handoff of a Collection to tertiary portals. Expand the current prototype for handoff to FireCloud.
```
mention above in the registry to link the two concepts.

theathorn replied:
Can you clarify what you mean here?

briandoconnor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,82 @@
+* Full authN/Z if the HCA stores controlled access data in the future. Leverage AuthN/Z from Data Store.
+* Browser interface with DSS Collections service to save/view/list/share/delete the query results from a search operation
+* Handoff of a Collection to tertiary portals. Expand the current prototype for handoff to FireCloud.
+* Save/view/share/update/delete search queries
```
and personal collections

theathorn replied:
done

brianraymor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,82 @@
+* HumanCellAtlas/data-browser-dev : data browser development discussions
+### Mailing list(s)
+### Discussion Forum(s)
+
```
If there are no **Mailing lists(s)** or **Discussion Forum(s)**, then these can be deleted.

theathorn replied:
removed.

theathorn committed:
Update charter.md

brianraymor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+The Data Portal provides the overall web-site for the HCA DCP with entry point from https://www.humancellatlas.org/. It provides the user experience for scientific and general users to upload, explore, search and run analyses on human genomic data. The Data Browser ("Explore") application provides a faceted search capability to find and download subsets of this data.
+
+## Objective
+Implementation of the Data Coordination Platform (DCP) website experience. Provide the access point for scientific and general users to search all HCA genetic data in order to identify a subset of the data for download and/or further analysis, providing file/bundle search on multiple clouds at PB scale, with strong public APIs.
```
*Minor:*
* Even Amazon _spells out_ PB as either _petabyte scale_ or _petabyte-scale_. 
* What is meant by _strong public APIs_?

mweiden commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+
+
+## Description
+The Data Portal provides the overall web-site for the HCA DCP with entry point from https://www.humancellatlas.org/. It provides the user experience for scientific and general users to upload, explore, search and run analyses on human genomic data. The Data Browser ("Explore") application provides a faceted search capability to find and download subsets of this data.
```
The first sentence, specifically the description `overall web-site`, is not clear. How about something like `The Data Portal is a landing website for the Human Cell Atlas (HCA) Data Coordination Platform (DCP). It is hosted on the main [HCA website](https://www.humancellatlas.org/).`?

mweiden commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+
+## In-scope
+### General
+* UX, visual and branding design with input from HCA leadership (GG, scientific steering committee, etc.)
```
I would define the acronym `GG` as it will likely be unfamiliar to new visitors.

brianraymor replied:
The DSC - *scientific steering committee* dissolved last night.  @jonahcool - is the GG committed to offering input on _UX, visual and branding design_ ?  @theathorn - we're also calling out work items that require oversight from Science governance under **Scientific "guardrails"** in the template. It flags items that may need more review by Science PMs. Is there a specific schedule or process that you have in mind - for example, does the planned October demo to the GG cover this situation?

mweiden commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* Production operation of the overall Data Portal web-site (see exclusions in Out-of-scope section)
+* Implementation performance scaling
+* DevSecOps compliance as required by the HCA Security Team
+* DevSecOps features for FISMA moderate deployments (authZ/N, logging, auditing, etc) by leveraging other projects (if this becomes a future requirement)
```
Change to `authentication and authorization`. This will be clearer to new members of the community than `authZ/N`.

mweiden commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+### General
+* UX, visual and branding design with input from HCA leadership (GG, scientific steering committee, etc.)
+* Participation in the HCA-wide standards for Continuous Integration (CI), Continuous Delivery (CD), system logging via Google Analytics, etc.
+* Production operation of the overall Data Portal web-site (see exclusions in Out-of-scope section)
```
`website`. Source https://en.wikipedia.org/wiki/Website

mweiden commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* Browser UI to search the DSS by specifying multiple search criteria (“facets”)
+* Download selected files from the DSS via the UI and/or HCA cli (the HCA cli itself is out-of-scope for this charter)
+* AuthN for tracking users' collections, saved searches, etc.
+* Full authN/Z if the HCA stores controlled access data in the future. Leverage AuthN/Z from Data Store.
```
`authentication and authorization`

mweiden commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* Integration with Vanilla Forums (discussion forum for developers)
+* Integration with Canny (to capture user feedback)
+### Data Browser
+* Create (or use Data Store provided) indexes as necessary for files in the Data Store, using public DSS APIs to read the metadata
```
This is the first mention of the Data Storage Service. Define acronym here as `Data Storage Service (DSS)` for later use.

mweiden commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* Download selected files from the DSS via the UI and/or HCA cli (the HCA cli itself is out-of-scope for this charter)
+* AuthN for tracking users' collections, saved searches, etc.
+* Full authN/Z if the HCA stores controlled access data in the future. Leverage AuthN/Z from Data Store.
+* Browser interface with DSS Collections service to save/view/list/share/delete the query results from a search operation
```
nit: why not `save, view, list, share, and delete`"?

mweiden commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* Full authN/Z if the HCA stores controlled access data in the future. Leverage AuthN/Z from Data Store.
+* Browser interface with DSS Collections service to save/view/list/share/delete the query results from a search operation
+* Handoff of a Collection of search results to a tertiary portal. Provide for direct handoff of search results to FireCloud.
+* Save/view/share/update/delete search queries and personal collections
```
See above.

mweiden commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+
+## Out-of-scope
+* Production operation of other HCA Data Coordination Platform (DCP) components in AWS (e.g. Data Store, Ingest, Secondary Analysis, Tertiary Analysis)
+* Contribute/Ingest UI (from EBI)
```
Separate these items and give a couple words more context. If I were a new user, I would not know what either of these things were.

mweiden commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+## Out-of-scope
+* Production operation of other HCA Data Coordination Platform (DCP) components in AWS (e.g. Data Store, Ingest, Secondary Analysis, Tertiary Analysis)
+* Contribute/Ingest UI (from EBI)
+* Methods Registry providing manage/search/filter/sort visualization and methods packages in the tertiary portal (from Broad)
```
nit: Why not something like `management, search, filtering, sorting, visualization, and methods ...`?

mweiden commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* HumanCellAtlas/data-portal : general data portal discussions
+* HumanCellAtlas/data-portal-dev : data portal development discussions
+* HumanCellAtlas/content-team : data portal content development discussions
+* HumanCellAtlas/data-browser-dev : data browser development discussions
```
nit: eliminate the extra space

brianraymor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* Browser support for multiple metadata schemas that are supported by the schema access library maintained by the Ingest service
+* Search views of Specimens, Projects and Files
+* Browser UI to search the DSS by specifying multiple search criteria (“facets”)
+* Download selected files from the DSS via the UI and/or HCA cli (the HCA cli itself is out-of-scope for this charter)
```
_or HCA cli (the HCA cli itself is out-of-scope for this charter)_ then this reference should be moved from here to the out-of-scope section.

brianraymor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+## In-scope
+### General
+* UX, visual and branding design with input from HCA leadership (GG, scientific steering committee, etc.)
+* Participation in the HCA-wide standards for Continuous Integration (CI), Continuous Delivery (CD), system logging via Google Analytics, etc.
```
What is meant by _Participation in ... standards_? (It could read like you're helping to develop a standard for example) 

I would also probably use the phrase _HCA DCP community standards_ or _best practices_.

brianraymor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* Frequent updates of summary statistics (cells, organs, donors, projects, labs, etc.) on Home Page
+* Coordinate development of user guide content (e.g. “Learn”) from multiple contributors
+* Coordinate development of developer documentation content (e.g. “Develop”) from multiple contributors. Includes API documentation and development guides.
+* Click through to Data Browser from search box and human body image
```
Some items in the charter read like feature descriptions  (_click through ..._). What's the **big picture** scope?

theathorn replied:
Attempted to retain only high-level descriptions of planned user functionality.

theathorn committed:
Incorporate feedback from Brian Raymor and Matt Weiden

brianraymor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* Search views of Specimens, Projects and Files
+* Browser UI to search the DSS by specifying multiple search criteria (“facets”)
+* Download selected files from the DSS via the UI and/or HCA cli
+* Full authentication and authorization if the HCA stores controlled access data in the future. Leverage the authetication and authorization implementation from the Data Store.
```
_future_ means out-of-scope for this charter?

theathorn replied:
Removed future items that are not yet committed on the roadmap.

brianraymor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+## In-scope
+### General
+* UX, visual, and branding design with input from the HCA brand working group
+* Participation in the HCA DCP community standards and best practices for Continuous Integration (CI), Continuous Delivery (CD), system logging via Google Analytics, etc.
```
I didn't see a response to my earlier question. Looking forward, community reviewers may become frustrated if their feedback is not addressed.

And I suggested -or- rather than -and-:

_What is meant by Participation in ... standards? (It could read like you're helping to develop a standard for example)_

_I would also probably use the phrase HCA DCP community standards **or** best practices._

theathorn replied:
Changed to "Adhere to ... best practices".

brianraymor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* API Documentation for portable, cloud-neutral APIs
+
+## Out-of-scope
+* Production operation of other HCA DCP components in AWS (e.g. Data Store Service, data ingestion process, Secondary Analysis workflows, Tertiary Analysis portals, etc.)
```
Why is it significant to call out this _production operation_ case?

theathorn replied:
Trying to convey that this charter does not cover DevOps operation of the other boxes (purple/ingest, blue/data store, green/secondary analysis, red/tertiary portals). Is there a better way to clarify this?

brianraymor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* Production operation of other HCA DCP components in AWS (e.g. Data Store Service, data ingestion process, Secondary Analysis workflows, Tertiary Analysis portals, etc.)
+* Any separate data ingest application developed by The European Bioinformatics Insitute (EBI) which may replace or augment the "Contribute" tab in the Data Portal
+* Methods Registry providing management, searching, filtering, sorting, visualization, and methods packages in the tertiary portal (registry is developed and maintained by the Broad Institute)
+* The HCA cli implementation
```
Please use _HCA DCP Command Line Interface (CLI)_ and what specifically is out-of-scope - the infrastructure implementation that's part of the Data Store charter?

theathorn replied:
yes - done.

brianraymor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+
+## Out-of-scope
+* Production operation of other HCA DCP components in AWS (e.g. Data Store Service, data ingestion process, Secondary Analysis workflows, Tertiary Analysis portals, etc.)
+* Any separate data ingest application developed by The European Bioinformatics Insitute (EBI) which may replace or augment the "Contribute" tab in the Data Portal
```
Insitute -> Institute

theathorn replied:
done.

brianraymor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* Browser interface to the DSS Collections service to save, view, list, share and delete the query results from a search operation
+* Handoff of a Collection of search results to a tertiary portal. Provide for direct handoff of search results to FireCloud.
+* Facilities to save, view, share, update, and delete search queries and personal collections on a per-user basis
+* Search results handoff to the Matrix service
```
Is there a pointer to the current plan of record related to the Matrix service and portal?

theathorn replied:
See https://docs.google.com/document/d/12gfF3UEgRZ0YJiBoScLMtcW6Cthv0_MZHQwj4WMYKPA/edit#heading=h.gk7s93eg9jfq and https://app.zenhub.com/workspace/o/humancellatlas/data-browser/issues/9

theathorn committed:
Incorporate further feedback from Brian Raymor

brianraymor commented on charters/DataPortal/charter.md:
```diff
@@ -34,25 +33,26 @@ Implementation of the DCP website experience. Provide the access point for scien
 * Nov 2018:  Present Pilot phase portal to community
 * EOY 2018:  Improved scaling/hardening for additional gaps identified in HCA DCP Pilot phase
+* 1H  2019:  Save search querys and query results (using DSS Collections APIs). Handoff of Collections to the Matrix service and to tertiary portals. Vanilla Forums support. Canny support.
 
```
querys-> queries

brianraymor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* Oct 2018:  Deploy MVP feature set as part of HCA DCP Pilot phase
+* Nov 2018:  Present Pilot phase portal to community
+* EOY 2018:  Improved scaling/hardening for additional gaps identified in HCA DCP Pilot phase
+* 1H  2019:  Save search querys and query results (using DSS Collections APIs). Handoff of Collections to the Matrix service and to tertiary portals. Vanilla Forums support. Canny support.
```
querys -> queries

brianraymor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* Browser interface to the DSS Collections service to save, view, list, share and delete the query results from a search operation
+* Handoff of a Collection of search results to a tertiary portal. Provide for direct handoff of search results to FireCloud.
+* Facilities to save, view, share, update, and delete search queries and personal collections on a per-user basis
+* Search results handoff to the Matrix service
```
Would this also be _Handoff of a **Collection** of search results_ similar to the tertiary portal above?

briandoconnor replied:
ya, I think anything that says "search results handoff..." should be "search results/collections handoff..."

theathorn committed:
Minor corrections

Fix spelling of "queries".
Make "Handoff of a Collection..." terminology consistent.

briandoconnor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* API Documentation for portable, cloud-neutral APIs
+
+## Out-of-scope
+* Production operation of other HCA DCP components in AWS (e.g. the DSS, data ingestion process, Secondary Analysis workflows, Tertiary Analysis portals, etc.)
```
I think what you're pointing out here is that we're not responsible for things linked to from the Data Portal.  Not a general statement that we don't operate other HCA DCP components.  Right?

theathorn replied:
Attempted to clarify further.

briandoconnor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+### Technical Lead
+Portal: [Dave Rogers](mailto:dave@clevercanary.com)
+Browser: [Hannes Schmidt](mailto:hannes@ucsc.edu)
+
```
Do you want to mention the overall dev team size this charter is scoped for?  That was something we discussed in the UCSC arch meeting today.

briandoconnor replied:
Related to this... is there a place in the charter that references the time period you think the charter will be valid for?  Are we going for, say, a charter that should be generally accurate for a year and then plan to revise the charter yearly?

brianraymor replied:
@briandoconnor - Both questions were answered inline - https://github.com/HumanCellAtlas/dcp-community/pulls#issuecomment-412667983

kozbo commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+The Data Portal is a landing site for the Human Cell Atlas (HCA) Data Coordination Platform (DCP). It is hosted at https://www.data.humancellatlas.org/ with an access link from the main HCA website at https://www.humancellatlas.org/. It provides the user experience for scientific and general users to upload, explore, search, and run analyses on human genomic data. The Data Browser ("Explore") app ...
+
+## Objective
+Implementation of the DCP website experience. Provide the access point for scientific and general users to search all HCA genetic data in order to identify a subset of the data for download and/or further analysis, providing file/bundle search on multiple clouds at petabyte scale with public APIs.
```
"Implementation of the DCP website experience." is a sentence fragment. Suggest 
`The objective of the Data Portal is to provide the DCP website implementation.`

theathorn replied:
done

kozbo commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* UX, visual, and branding design with input from the HCA brand working group
+* Adhere to the HCA DCP community best practices for Continuous Integration (CI), Continuous Delivery (CD), system logging via Google Analytics, etc.
+* Production operation of the overall Data Portal website in Amazon Web Services (AWS), with exclusions listed in the out-of-scope section
+* Implementation performance scaling
```
Do we know that there is performance scaling work to be done? Is this a requirement instead?

theathorn replied:
We don't know for sure that there is work to be done.
Changed to say "meet the performance scaling objectives of the DCP".

kozbo commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+   * Support plan for escalation of user reported bugs
+### Data Portal
+* Select and deploy a content management system
+* Integrate Data Browser and Data Portal UIs into a seamless UX and provide link to other groups' applications (e.g. a Data Ingestion UI)
```
nit: (e.g. **the** Data Ingestion UI)

theathorn replied:
done

kozbo commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* Create (or use Data Storage Service [DSS] provided) indexes to access files in the Data Store by using public DSS APIs to read the metadata
+* Browser support for multiple metadata schemas that are supported by the schema access library maintained by the Ingest service
+* Search views of Specimens, Projects and Files
+* Browser UI to search the DSS by specifying multiple search criteria (“facets”)
```
This seems a little terse. How about `Support to specify multiple search criteria ("facets") through the Browser UI.`

theathorn replied:
done

kozbo commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* Search views of Specimens, Projects and Files
+* Browser UI to search the DSS by specifying multiple search criteria (“facets”)
+* Download selected files from the DSS via the UI and/or HCA DCP Command Line Interface (CLI)
+* Leverage the authentication and authorization implementation from the Data Store
```
How about `Leverage the authentication and authorization libraries from the DevSecOps team to implement login and access control for the Data Portal and Data Browser.`

theathorn replied:
done and moved to General section.

kozbo commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* Browser UI to search the DSS by specifying multiple search criteria (“facets”)
+* Download selected files from the DSS via the UI and/or HCA DCP Command Line Interface (CLI)
+* Leverage the authentication and authorization implementation from the Data Store
+* Browser interface to the DSS Collections service to save, view, list, share and delete the query results from a search operation
```
Note that I replaced all DSS instances with Data Store, and only added the word "service" when I was specifically speaking about the service. We should probably be consistent across charters.

theathorn replied:
agree.

kozbo commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+* Production operation of other HCA DCP components in AWS (e.g. the DSS, data ingestion process, Secondary Analysis workflows, Tertiary Analysis portals, etc.)
+* Any separate data ingest application developed by The European Bioinformatics Institute (EBI) which may replace or augment the "Contribute" tab in the Data Portal
+* Methods Registry providing management, searching, filtering, sorting, visualization, and methods packages in the tertiary portal (registry is developed and maintained by the Broad Institute)
+* The HCA DCP CLI infrastructure implementation (which is part of the DSS)
```
I don't think that the community will expect the Data Portal group to be responsible for the CLI. I would vote to cut this "out of scope" item.

theathorn replied:
done

kozbo commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,81 @@
+## Milestones
+* Oct 2018:  Deploy MVP feature set as part of HCA DCP Pilot phase
+* Nov 2018:  Present Pilot phase portal to community
+* EOY 2018:  Improved scaling/hardening for additional gaps identified in HCA DCP Pilot phase
```
Again, do you know that you have scaling issues to solve? If so then ignore this comment.

theathorn replied:
clarified text.

kozbo reviewed (commented):
See in-line review

theathorn committed:
Minor updates

Minor updates following review from Brian O'Connor & Kevin Osborn.

theathorn committed:
Minor update

Further clarifications for comments from Brian O'Connor and Kevin Osborn.

malloryfreeberg commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,80 @@
+
+## Out-of-scope
+* Production operation of applications and services linked to from the Data Portal (e.g. the DSS, data ingestion service, Secondary Analysis workflows, Tertiary Analysis portals, etc.)
+* Any separate data ingest application developed by The European Bioinformatics Institute (EBI) which may replace or augment the "Contribute" tab in the Data Portal
```
It seems a bit odd to mention the specific Institutes of the DCP in the charter (EBI and Broad).

theathorn replied:
Removed.

malloryfreeberg commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,80 @@
+* API Documentation for portable, cloud-neutral APIs
+
+## Out-of-scope
+* Production operation of applications and services linked to from the Data Portal (e.g. the DSS, data ingestion service, Secondary Analysis workflows, Tertiary Analysis portals, etc.)
```
Do we have common, consistent language for all the DCP services? e.g. "Data Store Service" instead of "DSS", "Ingestion Service" instead of "data ingestion service, "Upload Service", etc.

mweiden replied:
I would follow the language used in the module charters. So these should be used consistently.
* `Data Storage Service (DSS)`
* `Ingestion Infrastructure`
* `Upload Service`

Ideally, the charters would link to each other.

theathorn replied:
Standardized terminology.

lauraclarke replied:
There has been some other shifts since this comment was made, I think once all the charters exist one person needs to check language consistency

malloryfreeberg commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,80 @@
+* Registration form for tertiary portals and list of portals available with click through to hosting web site (e.g. Dockstore)
+* Integration with Zendesk help desk
+* Integration with Vanilla Forums (discussion forum for developers)
+* Integration with Canny (to capture user feedback)
```
Is it appropriate to mention the specific applications (Zendesk, Vanilla Forums, Canny) for these features? Or is it better to not mention the implementation and just say that the Portal will have "Integration with help desk, discussion forum, and user feedback applications" or something similar.

theathorn replied:
good idea - done.

brianraymor replied:
If appropriate, you may want to call out your discussion forum for developers in **Communication**?

theathorn committed:
Terminology updates

Standardize on terms such as Ingestion Service, Upload Service, Data Storage Service, Data Processing Pipelines, Analysis Applications.
Remove references to specific Institutes.
Remove references to specific 3rd party product names and replace with generic terms.

brianraymor committed:
Updates to section headings

The charter template was updated - #18

theathorn committed:
"Handoff" changed to "interface" to the Matrix Service

lauraclarke commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,80 @@
+The Data Portal is a landing site for the Human Cell Atlas (HCA) Data Coordination Platform (DCP). It is hosted at https://www.data.humancellatlas.org/ with an access link from the main HCA website at https://www.humancellatlas.org/. It provides the user experience for scientific and general users to upload, explore, search, and run analyses on human genomic data. The Data Browser ("Explore") app ...
+
+## Objective
+The objective of the Data Portal is to provide the DCP website implementation. Provide the access point for scientific and general users to search all HCA genetic data in order to identify a subset of the data for download and/or further analysis, providing file/bundle search on multiple clouds at petabyte scale with public APIs.
```
Is there a difference between scientific users and general users? is it worth making that distinction?

lauraclarke replied:
Also, the HCA won't just distribute genetic data. I would remove that word. There will be proteomics, immunohistochemistry and maybe even MRIs plus whatever else gets thought up in the next few years.

If you want some examples I would change it to something like

"search all HCA data such as scRNA-seq and imaging transcriptomic data"

But removing the word genetic works too

lauraclarke commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,80 @@
+
+## In-scope
+### General
+* UX, visual, and branding design with input from the HCA brand working group
```
It would be good to add a definitions section and define what is meant by UX and UI, they don't seem to be defined anywhere

lauraclarke commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,80 @@
+
+## Description
+The Data Portal is a landing site for the Human Cell Atlas (HCA) Data Coordination Platform (DCP). It is hosted at https://www.data.humancellatlas.org/ with an access link from the main HCA website at https://www.humancellatlas.org/. It provides the user experience for scientific and general users to upload, explore, search, and run analyses on human genomic data. The Data Browser ("Explore") app ...
+
```
There seems to be no concept of ensuring the community can utilise the HCA data to deliver scientific value. This seems to be a significant oversight

malloryfreeberg replied:
> on human genomic data.

Perhaps this should be worded more like the data portal homepage and say "single-cell data" or "cellular resolution data". Currently, the DCP *does* accept mouse data and while we accept mostly genomic data now, in the future we might also accept proteomic and other non-genomic data.

lauraclarke replied:
Cellular resolution data seems like a good plan

simonjupp replied:
This is a bit misleading as the data portal itself will not be handling the user upload experience. Also, will it really be possible to run analyses from the dataportal?

theathorn replied:
I've adopted the "cellular  resolution data" suggestion and removed the references to "upload" and "running analyses".
I'm open to suggestions as to how to express the concept of "ensuring the community can utilise the HCA data to deliver scientific value".

lauraclarke replied:
Both the metadata and ingest support charters have statements in the objects like this 

`Ensuring the defined metadata schema provides a positive user experience for those who read and write them`

`Working with the UX team to ensure the submission process provides a positive user experience to data contributors`

These provide explicitly statements that usability is at the forefront of our concerns rather than an after thought

theathorn replied:
Updated Objectives.

lauraclarke commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,80 @@
+* Integration with Canny (to capture user feedback)
+### Data Browser
+* Create (or use Data Storage Service [DSS] provided) indexes to access files in the Data Store by using public DSS APIs to read the metadata
+* Browser support for multiple metadata schemas that are supported by the schema access library maintained by the Ingest service
```
@tburdett @theathorn @simonjupp @briandoconnor This looks like something that needs to be discussed as I am not sure @theathorn @briandoconnor we have a shared understanding of what metadata handing software ingest will provide to everyone

lauraclarke commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,80 @@
+* Support for specifying multiple search criteria (“facets”) through the Browser UI
+* Download selected files from the Data Store via the UI and/or HCA DCP Command Line Interface (CLI)
+* Browser interface to the Date Store Collections service to save, view, list, share and delete the query results from a search operation
+* Search results / Collections handoff to a tertiary portal. Provide for direct handoff of search results to FireCloud.
```
It might be better to make this third-party analysis services rather than specifically Firecloud as other people will likely want to integrate and it would seem a shame to explicitly limit to one service

lauraclarke commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,80 @@
+* Production operation of applications and services linked to from the Data Portal (e.g. the DSS, data ingestion service, Secondary Analysis workflows, Tertiary Analysis portals, etc.)
+* Any separate data ingest application developed by The European Bioinformatics Institute (EBI) which may replace or augment the "Contribute" tab in the Data Portal
+* Methods Registry providing management, searching, filtering, sorting, visualization, and methods packages in the tertiary portal (registry is developed and maintained by the Broad Institute)
+* Primary interface for collecting user feedback and feature/bug requests
```
This seems like a bad thing to be out of scope for the data portal. 

The data portal will be the first portal of call for most users of the HCA for the DCP which means it makes most sense for it to be the primary location for leaving feedback. I know @gabsie and @adriennes have thought about this

malloryfreeberg commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,78 @@
+* Coordinate development of developer documentation content (e.g. “Develop”) from multiple contributors. Includes API documentation and development guides.
+* Click through to Data Browser from search box and human body image
+* Registration form for tertiary portals and list of portals available with click through to hosting web site (e.g. Dockstore)
+* Integration with help desk, discussion forum, and user feedback applications
```
The Data Portal will also offer click through to the data submission UI from the Contribute section (though it doesn't at the moment).

theathorn replied:
Added to In-Scope section and to Milestones with a 1H 2019 delivery.

malloryfreeberg commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,78 @@
+* Download of selected files from the Data Store via the UI and/or HCA DCP Command Line Interface (CLI)
+* Browser interface to the Date Store Collections Service to save, view, list, share and delete the query results from a search operation
+* Search results / Collections interface to Analysis Applications. Provide UI for handoff of search results to FireCloud.
+* Facilities to save, view, share, update, and delete search queries and personal Collections on a per-user basis
```
How is "Facilities to save, view, share, update, and delete search queries and personal Collections on a per-user basis" different from "Browser interface to the Date Store Collections Service to save, view, list, share and delete the query results from a search operation"?

theathorn replied:
Removed the "personal Collections" clause. One facility is saving the search query (which can be used later on different data sets and the other is making an (immutable) Collection of search results.

malloryfreeberg commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,78 @@
+* API Documentation for portable, cloud-neutral APIs
+
+## Out-of-scope
+* Production operation of applications and services linked to from the Data Portal (e.g. the Data Storage Service, Ingestion Infrastructure, Upload Service, Data Processing Pipelines, Analysis Applications, etc.)
```
s/Ingestion Infrastructure/Ingestion Service/

Make sure these service names align with the standard terminology the other Charters are using.

theathorn replied:
OK.

malloryfreeberg commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,78 @@
+* Create (or use Data Storage Service [DSS] provided) indexes to access files in the Data Store by using public DSS APIs to read the metadata
+* Browser support for multiple metadata schemas that are supported by the schema access library maintained by the Ingestion Service
+* Search views of Specimens, Projects and Files
+* Support for specifying multiple search criteria (“facets”) through the Browser UI
```
Replace the formatted quotes with markdown quote (probably just a side effect of copy/paste).

theathorn replied:
What is a "markdown quote"? The current quote marks show up as expected when viewing the document.

malloryfreeberg replied:
“facets” (formatted or smart quotes, notice the slant) versus "facets" (the quotes that appear when you type them in the markdown document, I think they might also be called straight quotes).  Sometimes browsers and web services can't render the formatted/smart quotes. I know they look fine in the markdown document now, but if we ever ported this text to the Data Portal or another web service, they might cause issues.

malloryfreeberg commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,78 @@
+* HumanCellAtlas/data-portal: general data portal discussions
+* HumanCellAtlas/data-portal-dev: data portal development discussions
+* HumanCellAtlas/content-team: data portal content development discussions
+* HumanCellAtlas/data-browser-dev: data browser development discussions
```
Is data-browser-dev a closed channel? Will there be an open, non-dev, channel like for the Data Portal but for the Data Browser, where people can ask general questions about it?

theathorn replied:
Working on making data-browser-dev a public channel. data-browser is already a public channel.

malloryfreeberg replied:
I don't see #data-browser in the HCA Slack group. Is it perhaps called something different? I think it's ok if #data-browser-dev is meant to be a closed channel for developers only. I just can't see any channels with "browser" in the name in the HCA Slack group (except #file-browser), and it would be good to have at least 1 that is public or that at least other DCP members can be a part of.

malloryfreeberg commented:
I'm a bit confused about the Portal versus the Browser. This document is specifically titled to be for the Data Portal, but it also has significant content about the Browser. Will there be a separate Data Browser Charter? Or is the Data Browser considered such a crucial or the main part of the Portal that the two are going to be covered by the same Charter?

lauraclarke reviewed (changes requested):
The issues I have raised need to be addressed before this can be approved

malloryfreeberg reviewed (changes requested):
I have left some comments requesting changes. Happy to discuss them.

simonjupp commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,78 @@
+* Integration with help desk, discussion forum, and user feedback applications
+### Data Browser
+* Create (or use Data Storage Service [DSS] provided) indexes to access files in the Data Store by using public DSS APIs to read the metadata
+* Browser support for multiple metadata schemas that are supported by the schema access library maintained by the Ingestion Service
```
The schema access library is not part of the ingestion service charter

theathorn replied:
Should it be? I thought that Ingestion has agreed to maintain the metadata API library. Shall I remove the "maintained by" clause entirely?

tburdett replied:
The metadata API library will definitely not be maintained by the ingestion service. It's totally out of scope given that ingest is metadata agnostic. There _was_ some discussion about whether the metadata schema team (also partially at EBI, hence perhaps the confusion) should maintain this library that AFAIK hadn't reached any conclusion, but @lauraclarke can probably say more. Personally I think asking the metadata schema team to maintain any client library (let alone taking over one developed by someone else) represents a significant scope change that needs a lot more discussion.

The choice of library is an implementation detail anyway. If the data browser wants to make support for multiple schemas out of scope, you'll need to explain how you plan to only display the subset of data in the datastore that matches the current browser expectations. This is not something ingest can help with _by design:_ ingest has no ability to retain copies of old/legacy data.

theathorn committed:
Incorporate feedback from Laura, Mallory and Simon.

theathorn commented:
> I'm a bit confused about the Portal versus the Browser. This document is specifically titled to be for the Data Portal, but it also has significant content about the Browser. Will there be a separate Data Browser Charter? Or is the Data Browser considered such a crucial or the main part of the Portal that the two are going to be covered by the same Charter?

I think they should be one charter as they combine to provide the overall UI for the Portal. For example the Home Page of the Portal has statistics and a search bar that link directly to the Browser.

tburdett commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,79 @@
+## In-scope
+### General
+* UX, visual, and branding design with input from the HCA brand working group
+* Adhere to the HCA DCP community best practices for Continuous Integration (CI), Continuous Delivery (CD), and system logging via Google Analytics, etc.
```
Does this need calling out? If we have DCP community best practices, these should be published somewhere (AFAIK we don't have this). It would then be superfluous for all charters to reference this.

tburdett commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,79 @@
+* UX, visual, and branding design with input from the HCA brand working group
+* Adhere to the HCA DCP community best practices for Continuous Integration (CI), Continuous Delivery (CD), and system logging via Google Analytics, etc.
+* Production operation of the overall Data Portal website in Amazon Web Services (AWS), with exclusions listed in the out-of-scope section
+* Meet the performance scaling objectives of the DCP
```
This reads like the Data Portal won't be responsible for scaling objectives, that they will be handed out from someone else, rather than the data portal participating in the scale testing process.

I also think this should be a given rather than need explicitly stating

tburdett commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,79 @@
+* DevSecOps compliance as required by the HCA Security Team
+* Leverage the authentication and authorization implementation libraries from the DevSecOps team to implement login and access control for the Data Portal and Data Browser
+* Community engagement:
+   * Triage and integration of feature requests into the work queue 
```
Does this apply DCP wide? The Data Portal will be the first port of call for users submitting feature requests. Will they be triaged by the Portal team? If not, how do we expect users to submit feature requests for other components?

tburdett commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,79 @@
+   * Triage and integration of feature requests into the work queue 
+   * Support plan for escalation of user reported issues
+### Data Portal
+* Select and deploy a content management system
```
This is a milestone, not a scoping statement. Also this is probably implementation details

jodihir replied:
This has been done and the CMS is working out well now;  perhaps the point is more like "maintain a content management system that meets the needs of the portal as it develops".

tburdett commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,79 @@
+   * Support plan for escalation of user reported issues
+### Data Portal
+* Select and deploy a content management system
+* Integrate the Data Browser User Interface (UI) and Data Portal UI into a seamless UX and provide links to other groups' applications (e.g. the Ingestion Service UI)
```
This seems inconsistent with the earlier statement that the Portal
> provides the user experience (UX) for users to explore cellular resolution data

However, this statement seems to shift the "explore" functionality into the browser application with seamless UX. I think this is the correct way to represent it (portal provides access to many applications, including the browser) but this would logically suggest splitting charters

tburdett commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,79 @@
+### Data Portal
+* Select and deploy a content management system
+* Integrate the Data Browser User Interface (UI) and Data Portal UI into a seamless UX and provide links to other groups' applications (e.g. the Ingestion Service UI)
+* Frequent updates of summary statistics (cells, organs, donors, projects, labs, etc.) on the DCP home page
```
Can you provide more details about "frequent"? Live? Daily? Monthly?

tburdett commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,79 @@
+* Registration form for analysis portals and list of portals available with click through to hosting web site (e.g. Dockstore)
+* Integration with help desk, discussion forum, and user feedback applications
+### Data Browser
+* Create (or use Data Storage Service [DSS] provided) indexes to access files in the Data Store by using public DSS APIs to read the metadata
```
Create or use - I think you absolutely must pick one here. The difference in approach massively impacts scope. 

DSS will necessarily contain a very large diversity of content, some of which may be old, partially accurate, low value, or simply so niche that it is impossible to display in a way that matches the visual paradigm for all other DCP contents. If you reuse DSS indexes, the data browser will have to handle this dynamically at viz time. If you create your own indexes, you can choose to index only those contents that can easily be displayed and flag the rest

tburdett commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,79 @@
+### Data Browser
+* Create (or use Data Storage Service [DSS] provided) indexes to access files in the Data Store by using public DSS APIs to read the metadata
+* Browser support for multiple metadata schemas that are supported by the schema access library maintained by the Ingestion Service
+* Search views of Specimens, Projects and Files
```
Projects and Files are primary concepts in the metadata domain model whereas Specimen is a submittable type implementation and therefore represents an assumption that might not be safe. This might be fine as long as you are prepared to filter out bundles without specimens (although there are none of these yet)

tburdett commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,79 @@
+* Facilities to save, view, share, update, and delete search queries on a per-user basis
+* Facilities to utilize the Date Store Collections Service to save, view, list, share and delete the query results from a search operation
+* Search results / Collections interface to Analysis Applications
+* Search results / Collections interface to the Matrix Service
```
This makes me confused about the role of the DSS vs the role of the data browser. Which services use the browser and which use the DSS directly? And what are the requirements that drive this separation?

tburdett reviewed (changes requested):
Having added a comment and based on the reference to the Ingest Service, I felt I should probably leave a review. Sorry it's quite late! 

I agree that the separation between the Data Portal and Data Browser is very unclear. I think there's a natural separation here: the website that users will interact with that provides access to a number of applications, of which the Data Browser is one, Ingestion Service is another. It makes more sense to me to separate these; as a concrete example, Ingest will at some point want the UI to integrate with the Portal but doesn't really need to integrate with the Browser.

jodihir commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,79 @@
+* Integrate the Data Browser User Interface (UI) and Data Portal UI into a seamless UX and provide links to other groups' applications (e.g. the Ingestion Service UI)
+* Frequent updates of summary statistics (cells, organs, donors, projects, labs, etc.) on the DCP home page
+* Coordinate development of user guide content (e.g. “Learn”) from multiple contributors
+* Coordinate development of developer documentation content (e.g. “Develop”) from multiple contributors. Includes API documentation and development guides.
```
Add a point about coordinating wording on website pages that require written content but aren't specifically documentation (eg: About section)

theathorn committed:
Remove Data Browser charter

This charter is now specific to the Data Portal. A separate charter will be developed for the Data Browser,
The Data Portal charter has been substantially updated following extensive feedback.

theathorn committed:
Add milestone for authentication support.

theathorn committed:
Initial revision

theathorn committed:
Delete from this branch

tburdett commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,69 @@
+*Data Browser* A DCP application which provides a faceted search capability.
+
+## Objectives
+The objective of the Data Portal is to provide the DCP website implementation. It provides the access point for users to search all HCA genetic data in order to identify a subset of the data for download and/or further analysis at petabyte scale on public clouds.
```
Too narrow - access point to search, contribute to or learn about the Human Cell Atlas?

theathorn replied:
Agreed.

tburdett commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,69 @@
+### Core capabilities
+* Implement a UX, visual, and branding design with input from the HCA brand working group.
+* Maintain a content management system that meets the needs of Data Portal content developers.
+* Integrate the Data Browser User Interface (UI) and Data Portal UI into a seamless UX and provide links to other DCP applications (e.g. the Ingestion Service UI).
```
I think the distinction between browser and other applications is artificial. Why not just "Integrate DCP applications (e.g. the Data Browser, Ingestion Service) into a seamless UX with a common look and feel"?

theathorn replied:
Agreed.

tburdett commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,69 @@
+* Implement a UX, visual, and branding design with input from the HCA brand working group.
+* Maintain a content management system that meets the needs of Data Portal content developers.
+* Integrate the Data Browser User Interface (UI) and Data Portal UI into a seamless UX and provide links to other DCP applications (e.g. the Ingestion Service UI).
+* Provide for frequent (minimum daily) updates of summary statistics (cells, organs, donors, projects, etc.) on the DCP home page.
```
I'm unsure about the daily cadence here. My understanding of the portal was that it was designed to coordinate data releases to the community, whereas this suggests continuous release. Who is therefore responsible for the coordinated release of data to the community (i.e. ensuring all data is QC-ed to the same standard and all metadata upgraded to the same schema version)? Are you expecting the data browser / data store / query service will do this? Or do you really expect users of the data to do this?

theathorn replied:
This is just referencing the cadence at which the Portal will update the summary statistics displayed on the Portal home page which is independent of the ingestion of the data. So the stats won't be immediately up to date with the data but may lag by the minimum. In the implementation I think it's more often than once per day, but that is a detail.

The portal does not "coordinate data releases to the community" - it is the UI that shows users what is there, The other services ingest data, create and index bundles, etc. The Data Browser (separate charter) displays data according to the indexes created, and the Data Browser is one of the applications accessed via the Data Portal. QC of data and metadata upgrade are outside the scope of the Data Portal and Data Browser charters.

tburdett replied:
I think this comment is more related to the Browser than the Portal, but seeing as we started this thread I'll continue here...

Making it out of scope for the portal/browser is not unreasonable, but _someone_ has to coordinate releases. I would like to see impacted charters addressing the issue. The DCP will contain data that has been ingested over time, using different metadata standards, different analysis pipeline versions, different QC panels, etc. I'd expect the Data Portal / Browser (this charter is clearly the wrong place for this discussion following the split!) to either expect some sort of stamp that indicates the data is up to current standards, or else to provide this stamp themselves. Otherwise, you will face a considerable amount of engineering work to support display of all possible versions of every byte of data in the DSS, and pass the problem onto users when they download the resulting data. I have a suspicion this will be an enormous problem once imaging data starts to be ingested, for example.

lauraclarke replied:
It is worth noting that this will very likely fall under the remit of the data operations team. That being said, just as the ingestion service, metadata schema and data store charters contain reciprocal statements about how responsibilities for different angles of metadata specification and implementation are divided. I would assume the data browser/portal will need to be linked to these efforts as they will create display requirements both for static and browsable content.

kbergin replied:
Of note, the secondary analysis team thus far plans to reanalyze all data when a pipeline has a major version update.

theathorn committed:
Incorporate feedback from tburdett.

theathorn committed:
Minor wording change

Change "statistics on the DCP" to "statistics on the Data Portal home page".

theathorn committed:
Minor wording fix

analyzes => analyze

brianraymor commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,69 @@
+* HumanCellAtlas/data-portal-dev: data portal development discussions
+* HumanCellAtlas/content-team: data portal content development discussions
+
+## Github repositories
```
Can you add a brief description of the repo(s)?

theathorn replied:
Done.

brianraymor committed:
Editorial updates to Slack Channels

Adopt best practices for slack channel format

theathorn committed:
Add github repos descriptions.

lauraclarke commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,69 @@
+*Data Browser* A DCP application which provides a faceted search capability.
+
+## Objectives
+The objective of the Data Portal is to provide the DCP website implementation. It provides the access point for users to search, contribute to, or learn about, the HCA genetic data in order to identify a subset of the data for download and/or further analysis at petabyte scale on public clouds.
```
HCA genetic data is a very misleading statement

I would propose changing it to something like

`the cellular resolution and other data which is part of the HCA collection,`

theathorn replied:
Issues addressed with revised Description and Objectives sections. Does this look OK now?

lauraclarke commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,69 @@
+## In-scope
+
+### Core capabilities
+* Implement a UX, visual, and branding design with input from the HCA brand working group.
```
Is UX the right term here? do you mean UI?

gabsie replied:
>Implement a UX, visual, and branding design with input from the HCA brand working group.

I would probably say here, in two separate points:
 - Implement a visual design which addresses the needs and goals of the identified main user personas. 
It should therefore be driven by the user experience research insights and interaction design guidelines coming from the User Experience (UX) team. 
 - Implement a visual design which is in line with the Human Cell Atlas brand guidelines.

theathorn replied:
Updated the first two Core capabilities. Does this look OK now?

gabsie commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,69 @@
+
+
+## Description
+The Data Portal is the landing site for the Human Cell Atlas (HCA) Data Coordination Platform (DCP). It is hosted at https://www.data.humancellatlas.org/ with an access link from the main HCA website at https://www.humancellatlas.org/. It provides the user experience (UX) for users to explore cellular resolution data.
```
>It provides the user experience (UX) for users to explore cellular resolution data.

Hi, this sounds strange. Maybe a simpler way to phrase this: 

The main goal of the Data Portal is to provide life scientists interested in the Human Cell Atlas (HCA) with an easy and efficient way to explore and use cellular resolution data.

theathorn committed:
Modify Description and Objectives sections

Description: Replace "genetic data" with "cellular resolution data".
Objectives: Split objectives into separate statements. Emphasize goal of providing a positive user experience.

lauraclarke commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,73 @@
+* Implement a visual design which addresses the needs and goals of the identified main user personas. It is driven by the user experience research insights and interaction design guidelines provided by the User Experience (UX) team.
+* Implement a visual design which is in line with the HCA brand guidelines.
+* Maintain a content management system that meets the needs of Data Portal content developers.
+* Integrate DCP applications (e.g. the Data Browser, Ingestion Service, etc.) into a seamless UX with a common look and feel.
```
Again, I am not sure UX is the right term here. While you might mean user experience using the acronym UX carried with it certain connotations which aren't quite right in this context

theathorn replied:
Changed UX to UI.

lauraclarke commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,73 @@
+## In-scope
+
+### Core capabilities
+* Implement a visual design which addresses the needs and goals of the identified main user personas. It is driven by the user experience research insights and interaction design guidelines provided by the User Experience (UX) team.
```
You shouldn't say `It is driven by the user experience research` but you should state what you mean by It, if you mean the visual design, say the visual design, if you mean the UI, say the UI

theathorn replied:
Changed "It" to "The visual design".

theathorn committed:
Minor clarifications

"seamless UX" => "seamless UI".
"It is driven..." => "The visual design is driven..."

tburdett commented on charters/DataPortal/charter.md:
```diff
@@ -0,0 +1,73 @@
+
+
+## Description
+The Data Portal is the landing site for the Human Cell Atlas (HCA) Data Coordination Platform (DCP). It is hosted at https://www.data.humancellatlas.org/ with an access link from the main HCA website at https://www.humancellatlas.org/. It provides users interested in the HCA with an easy and efficient way to explore and use data at cellular resolution.
```
Is the url https://www.data.humancellatlas.org/ correct? I was sort of expecting https://data.humancellatlas.org. The currently deployed staging environment has no www; if it's the current deployment that's in error then this is fine.

theathorn replied:
Good catch - now fixed.

tburdett commented:
Thanks for the changes - this is much clearer now. Happy to approve, feel free to resolve the open comment on the URL if it's not relevant

theathorn committed:
Remove "www" from DCP website address
